stdlib/private/CMakeLists.txt

```
@@ -1,26 +1,30 @@
-if(SWIFT_BUILD_SDK_OVERLAY)
+if(SWIFT_BUILD_SDK_OVERLAY OR (SWIFT_BUILD_TEST_SUPPORT_MODULES AND (NOT SWIFT_BUILD_DYNAMIC_SDK_OVERLAY_default) AND (NOT (SWIFT_FREESTANDING_FLAVOR STREQUAL "apple"))))
```

Optional: consider breaking the condition on multiple lines for readability

```suggestion
if(SWIFT_BUILD_SDK_OVERLAY
   OR (SWIFT_BUILD_TEST_SUPPORT_MODULES
       AND (NOT SWIFT_BUILD_DYNAMIC_SDK_OVERLAY_default)
       AND (NOT (SWIFT_FREESTANDING_FLAVOR STREQUAL "apple"))))
```